src/models/user.ts

```diff
@@ -0,0 +1,92 @@
+// Import the necessary modules
+import mongoose, { Document, Schema } from 'mongoose';
+
+// Define the user schema
+interface UserSchema extends Document {
+  firstname: string;
+  lastname: string;
+  email: string;
+  phone: string;
+  profileImage: string;
+  password: string;
+  role: 'admin' | 'cook' | 'customer';
+  isActive: boolean;
+  address: {
+    city: string;
+    district: string;
+    neighborhood: string;
+    streetAddress: string;
+  };
+}
+
+
+const userSchema: Schema<UserSchema> = new Schema({
+  firstname: {
+    type: String,
+    required: [true,'First name is required']
+  },
+  lastname: {
+    type: String,
+    required: [true,'Last name is required']
+  },
+  email: {
+    type: String,
+    required: [true, 'Email is required'],
+    unique: true,
+    validate: {
+        validator: (value: string) => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value),
+        message: 'Invalid email format'
+    }
+  },
+  phone: {
+    type: String,
+    required: [true, 'Phone number is required'],
+    validate: {
+      validator: (value: string) => /^\d{10}$/.test(value),
+      message: 'Invalid phone number format'
+    }
```

Comment:
no need to implement complicated validator here in the schema, we will add a validation later for that later on.
schema should be simple as much as possible.
so i suggest deleting validate.